src/dsymbol.d

```diff
@@ -460,20 +460,20 @@ extern (C++) class Dsymbol : RootObject
     // instantiated speculatively, eg from inside is(typeof()).
     // Return the speculative template instance it is part of,
     // or NULL if not speculative.
-    final TemplateInstance isSpeculative()
+    final inout(TemplateInstance) isSpeculative() inout
     {
-        Dsymbol par = parent;
+        Dsymbol par = cast()parent;
```

Comment:
Would it be possible to remove the casts if we make the `par` and `ti` variable declarations `auto`?
Or does `isTemplateInstance` and `toParent` require mutable `this`?